.github/workflows/ci.yml

```diff
@@ -33,8 +33,10 @@ jobs:
           key: ${{ runner.OS }}-cargo-${{ hashFiles('**/Cargo.lock') }}
           restore-keys: |
             ${{ runner.OS }}-cargo-
-      - name: Build library
+      - name: Build library (default features)
         run: cargo build
+      - name: Build library (all features)
+        run: cargo build --all-features
```

Comment:
I wonder what a sensible CI strategy is here.

First about which feature combinations to test. Adding an all features build is I think a good idea, because it future-proofs us against forgetting to add new feature flags to CI. Testing individual featurescould also be useful, though, to ensure the features are independent or specify the correct dependencies.

Second about having build and test targets. If I understand the [docs](https://doc.rust-lang.org/cargo/commands/cargo-test.html#target-selection) correctly, test implies build. It might be enough to only have test commands (with the desired feature combinations).

Reply:
My proposal in 4438ad732bf7ec2ec541bc14116e36714bcae6bc: run `cargo test` with default and all features, and build the CLI individually as we did before.

Reply:
> Second about having build and test targets

I had these separate mostly so that we would see compilation failures from the library code itself earlier / in a separate section.  You're right that `cargo test` will build whatever other stuff is necessary, so if we're finding that separation not to be useful, I'm :+1: to how you've reduced the number of steps.

(Though note that keeping them separate should not increase the CI time, since the build results from the separate `cargo build` step will be reused during the `cargo test` step.)

Reply:
That's a good point, about the caching. I think I'll split them up after all then.